### PR TITLE
Add Brazilian Portuguese language support

### PR DESCRIPTION
**Pull request by joubertredrat**
This is imcomplete translation, missing finish on `Config Cheat Sheet` and `Hacking on Gitea`. I will finish this on this week.

Ping @andreynering for review

### CODE REVIEW

File: content/doc/advanced.pt-br.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+date: "2016-12-01T16:00:00+02:00"
+title: "Avançado"
+slug: "avancado"
```
Comment (andreynering) [inline content/doc/advanced.pt-br.md]: @tboerger Should other languages have translated slugs? I think not.

File: content/doc/advanced/config-cheat-sheet.en-us.md [hunk 2]
```diff
@@ -193,3 +193,7 @@ Note: Actually, Gitea supports only SMTP with STARTTLS.
 
 - `SHOW_FOOTER_BRANDING`: Enable this to show Gitea branding in the footer.
 - `SHOW_FOOTER_VERSION`: Enable this to show Gitea version information in the footer.
+
+## Anything missing?
```
Comment (andreynering) [inline content/doc/advanced/config-cheat-sheet.en-us.md]: Missing translation

Comment (tboerger) [inline content/doc/advanced/config-cheat-sheet.en-us.md:197]: If we want to add it to every page we should do that on theme level.

Comment (joubertredrat) [reply content/doc/advanced/config-cheat-sheet.en-us.md:197]: @tboerger I think that this is better if possible

Comment (lunny) [reply content/doc/advanced/config-cheat-sheet.en-us.md:197]: And add multi-forums support.

File: content/doc/advanced/config-cheat-sheet.pt-br.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
+---
+date: "2016-12-26T16:00:00+02:00"
+title: "Configuração"
+slug: "configuracao"
```
Comment (andreynering) [inline content/doc/advanced/config-cheat-sheet.pt-br.md:4]: @tboerger Should we translate slugs to other languages? I think we should not.

File: content/doc/advanced/config-cheat-sheet.pt-br.md [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+Antes de começar, certifique-se de que qualquer alteração deve ser feita em `custom/conf/app.ini` ao qualquer local correspondente.
+
+Todas as configurações padrão podem ser encontradas em [app.ini](https://github.com/go-gitea/gitea/blob/master/conf/app.ini). Se você ver qualquer coisa parecida com `%(X)s`, este é um recurso provido pelo [ini](https://github.com/go-ini/ini/#recursive-values) para leitura de valores recursivamente.
+
+Qualquer configuração que esteja marcado com :exclamation: significa que você deve manter o valor padrão, a menos que você entenda perfeitamente o que está fazendo.
+
+## Geral
+
+- `APP_NAME`: Nome da aplicação, altere para qualquer um que desejar.
+- `RUN_USER`: Usuário que está executando o Gitea, nós recomandamos que seja `git`; porém, altere isto para qualquer usuário se este está executando o Gitea em seu ambiente. Gitea pdoerá falhar se este valor não estiver definido corretamente.
```
Comment (andreynering) [inline content/doc/advanced/config-cheat-sheet.pt-br.md:29]: pdoerá -> poderá

File: content/doc/advanced/config-cheat-sheet.pt-br.md [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+- `HTTP_ADDR`: Endereço de escuta HTTP.
+- `HTTP_PORT`: Porta de escuta HTTP.
+- `DISABLE_SSH`: Desativa o recurso de SSH quando este não está disponível.
+- `START_SSH_SERVER`: Inicia o servidor SSH embutido quando ativado.
+- `SSH_PORT`: Porta SSH, caso não seja a porta `22`.
+- `OFFLINE_MODE`: Desativa o uso de CDN para arquivos estáticos e Gravatar para imagens de avatar.
+- `DISABLE_ROUTER_LOG`: Desativa a exibição de log do router.
+- `CERT_FILE`: Arquivo do certificado usado para HTTPS.
+- `KEY_FILE`: Arquivo da chave usado para HTTPS.
+- `STATIC_ROOT_PATH`: Upper level of template and static files path, default is the path where Gitea is located.
```
Comment (andreynering) [inline content/doc/advanced/config-cheat-sheet.pt-br.md:72]: Pasta onde se encontram as pastas `templates` e `public` do Gitea. O valor padrão é a pasta onde o executável do Gitea está localizado.

File: content/doc/advanced/config-cheat-sheet.pt-br.md [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+- `PATH`: Somente para SQLite3, pasta do arquivo de banco de dados.
+
+## Segurança (`security`)
+
+- `INSTALL_LOCK`: Indica se a página de instalação deve ser aberta (a configuração da conta de admintrador está envolvida, por conta disto é um valor importante).
+- `SECRET_KEY`: Chave secreta global para a segurança do seu servidor, **É preferível mudar** (será gerado uma chave aleatória sempre que você instalar).
+- `LOGIN_REMEMBER_DAYS`: Tempo de vida do Cookie em dias.
+- `COOKIE_USERNAME`: Nome do cookie que salva o nome de usuário.
+- `COOKIE_REMEMBER_NAME`: Nome do cookie que salva a informação de autologin.
+- `REVERSE_PROXY_AUTHENTICATION_USER`: Header name for reverse proxy authentication username.
```
Comment (andreynering) [inline content/doc/advanced/config-cheat-sheet.pt-br.md:93]: Nome do cabeçalho de autenticação de usuário, para proxy reverso.

File: content/doc/advanced/config-cheat-sheet.pt-br.md [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+
+## Serviço (`service`)
```
Comment (andreynering) [inline content/doc/advanced/config-cheat-sheet.pt-br.md:95]: Missing translations

File: content/doc/advanced/config-cheat-sheet.en-us.md (hunk 2 above)
Comment (tboerger) [reply content/doc/advanced/config-cheat-sheet.en-us.md:197]: Than let's add this to the theme based on an i18n string.

File: content/doc/installation/with-docker.pt-br.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  sidebar:
+    parent: "instalacao"
+    name: "Com Docker"
+    weight: 10
+    identifier: "instalacao-com-docker"
+---
+
+# Instalação com Docker
+
+Nós providenciamos automaticamente imagens de Docker atualizadas dentro da nossa organização de Hub Docker. It is up to you and your deployment to always use the latest stable tag or to use another service that updates the Docker image for you. Antes de tudo você deve simplesmente fazer o pull do container Docker:
```
Comment (andreynering) [inline content/doc/installation/with-docker.pt-br.md:18]: > It is up to you and your deployment to always use the latest stable tag or to use another service that updates the Docker image for you.

A decisão é sua de utilizar sempre a última tag estável ou usar algum serviço que atualize a imagem do Docker pra você.

**Review by andreynering — COMMENTED** [review]
Very good til now. Waiting for the rest of translations...

File: content/doc/advanced/config-cheat-sheet.pt-br.md (hunk 3 above)
Comment (tboerger) [reply content/doc/advanced/config-cheat-sheet.pt-br.md:4]: No, please keep the English version for slugs.

**Comment by tboerger** [comment]
@joubertredrat any progress?

**Comment by joubertredrat** [comment]
@tboerger I'm waiting finish to implement documentation in en-us to I make new PR this new changes

**Comment by avelino** [comment]
@joubertredrat the @andreynering made some comments in your translation, you reviewed all?

Need some help?

**Comment by strk** [comment]
I think we moved translations to crowdin, although I don't find references to that.
@lunny ?

**Comment by lunny** [comment]
@strk we moved gitea's translations to crowdin, but not this.

**Comment by strk** [comment]
@joubertredrat does en-us look fine now ?

**Comment by joubertredrat** [comment]
hi @strk I'm off about this. files was moved to crowind or translation will continue manual?

**Comment by avelino** [comment]
@lunny Let's use https://www.transifex.com/

**Comment by lunny** [comment]
@avelino it doesn't support ini format.

**Comment by tboerger** [comment]
The docs are not managed via any external service.

**Comment by joubertredrat** [comment]
Ok @tboerger I will provide this translation on this week

**Comment by pgaskin** [comment]
Any news?

**Comment by andreynering** [comment]
I think the docs are too much information to translate. Even if we did it, it would be outdated soon. 

This can be reopened if you change idea @jourbertradhat

**Pull request closed without merging**